src/main/java/io/github/shafthq/shaft/gui/image/ScreenshotManager.java

```diff
@@ -562,6 +563,10 @@ private static void startAnimatedGif(byte[] screenshot) {
                         + testCaseName + ".gif";
                 gifRelativePathWithFileName = SCREENSHOT_FOLDER_PATH + SCREENSHOT_FOLDER_NAME + gifFileName;
 
+                // get the width and height of the current window of the browser
+                var height = Integer.valueOf(new BrowserActions().getWindowHeight());
+                var width = Integer.valueOf(new BrowserActions().getWindowWidth());
```

Comment:
this works (Y)
//TODO: optimize, since we already have the browser width and height when we initialize the browser, so there is no need to re-get them at this point. it's costing us two webdriver calls, and in the near future we should focus on reducing the number of webdriver calls as they have significant impact on performance.